templates/flavors/default/machine-deployment.yaml

```diff
@@ -45,6 +45,8 @@ metadata:
 spec:
   template:
     spec:
+      kubeadmConfigSpec:
```

Comment:
added here

Reply:
you don't need kubeadmConfigSpec, it's already in spec
```suggestion
```

Reply:
for reference https://github.com/kubernetes-sigs/cluster-api/blob/master/bootstrap/kubeadm/api/v1alpha3/kubeadmconfigtemplate_types.go#L29